clinica/pipelines/pet/volume/utils.py

```diff
@@ -187,91 +183,79 @@ def pet_pvc_name(pet_image: str, pvc_method: str) -> str:
     """Normalize the provided `pet_image` by dividing by the mean
     value of the region defined by the provided `region_mask`.
 
     Parameters
     ----------
-    pet_image : str
-        Path to the Nifti1Image which should be normalized.
+    pet_image : Path
+        The path to the Nifti1Image which should be normalized.
 
-    region_mask : str
-        Path to the mask to be used to define the region.
+    region_mask : Path
+        The path to the mask to be used to define the region.
 
     Returns
     -------
-    suvr_pet_path : str
-        Path to the normalized Nifti1Image.
+    suvr_pet_path : Path
+        The path to the normalized Nifti1Image.
     """
-    from os import getcwd
-    from os.path import basename, join
-
-    import nibabel as nib
-    import numpy as np
-
     pet = nib.load(pet_image)
     ref = nib.load(region_mask)
-
     region = pet.get_fdata(dtype="float32") * ref.get_fdata(dtype="float32")
     region_mean = np.nanmean(np.where(region != 0, region, np.nan))
-
     data = pet.get_fdata(dtype="float32") / region_mean
-
-    suvr_pet_path = join(getcwd(), "suvr_" + basename(pet_image))
-
+    suvr_pet_path = Path.cwd() / f"suvr_{pet_image.name}"
     suvr_pet = nib.Nifti1Image(data, pet.affine, header=pet.header)
     nib.save(suvr_pet, suvr_pet_path)
 
     return suvr_pet_path
 
 
-def atlas_statistics(in_image: str, in_atlas_list: list) -> list:
+def compute_atlas_statistics(image: Path, atlas_names: List[str]) -> List[Path]:
     """Generate regional measure from atlas_list in TSV files.
 
     For each atlas name provided it calculates for the input image the mean
     for each region in the atlas and saves it to a TSV file.
 
     Parameters
     ----------
-    in_image : str
-        Path to the Nifti image.
+    image : Path
+        The path to the Nifti image.
 
-    in_atlas_list : List
+    atlas_names : List of str
         List of names of atlas to be applied.
 
     Returns
     -------
-    atlas_statistics : List
+    atlas_statistics : List of paths
         List of paths to TSV files.
     """
-    from pathlib import Path
-
-    from nipype.utils.filemanip import split_filename
-
+    from clinica.utils.filemanip import get_filename_no_ext
     from clinica.utils.statistics import statistics_on_atlas
 
-    orig_dir, base, ext = split_filename(str(in_image))
     atlas_statistics_list = []
-    for atlas in in_atlas_list:
-        out_atlas_statistics = Path.cwd() / f"{base}_space-{atlas}_statistics.tsv"
-        statistics_on_atlas(in_image, atlas, out_atlas_statistics)
+    for atlas in atlas_names:
+        out_atlas_statistics = (
+            Path.cwd() / f"{get_filename_no_ext(image)}_space-{atlas}_statistics.tsv"
+        )
+        statistics_on_atlas(image, atlas, out_atlas_statistics)
         atlas_statistics_list.append(out_atlas_statistics)
-        break
+        break  # Why is there a break here ????
```

Comment:
Feels like it would happen all the time ; is it not possible just to remove it ?

Reply:
It will indeed happen all the time and results in this function returning a list with a single value.
It doesn't make any sense and contradicts the documentation of the function, but I haven't checked whether calling code is relying on this single value list or not.
I suggest we investigate later (i.e. not in this PR) since the aim is only to refactor things.

Reply:
Alright then, the PR LGTM !

Reply:
Created #1228 to remind us about that.